### PR TITLE
Refactor render() in AppBody.jsx

### PR DESCRIPTION
**Pull request by giantelk**
Moved fugy todo list out side of the main return. In future could also put this in separate method or React component.

### CODE REVIEW

File: examples/react-todos/client/components/ListTodos.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                    </span> : "" }
+              </Link>
+            );
+          });
+
+    return (
+      <div>
+        { showTodoList }  
+      </div>
+      );
```
Comment (stubailo) [inline examples/react-todos/client/components/ListTodos.jsx]: Weird indentation

File: examples/react-todos/client/components/ListTodos.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+var Link = ReactRouter.Link;
+
+
+ListTodos = React.createClass({
+  render() {
+    var self = this;
+
+    var  showTodoList = this.props.lists.map(function (list) {
+            var className = "list-todo";
```
Comment (stubailo) [inline examples/react-todos/client/components/ListTodos.jsx]: This is indented too far

File: examples/react-todos/client/components/accounts/UserSidebarSection.jsx [hunk 3]
```diff
@@ -6,9 +6,11 @@ UserSidebarSection = React.createClass({
       menuOpen: false
     };
   },
+
```
Comment (stubailo) [inline examples/react-todos/client/components/accounts/UserSidebarSection.jsx]: Is there a good reason to have all of these blank lines? If you can show me a popular style guide somewhere that suggests having blank lines between methods, I'll go for it.

File: examples/react-todos/client/components/accounts/UserSidebarSection.jsx [hunk 4]
```diff
@@ -29,19 +35,30 @@ UserSidebarSection = React.createClass({
...
+        <div className="btns-group-vertical">
+          <a href="#" className="btn-secondary" onClick={ self.toggleMenuOpen }>
+            <span className={ arrowIconClass } />
+            { emailUsername }
+          </a>
+          { self.state.menuOpen ?
+            <a className="btn-secondary" onClick={ self.logout } >Logout</a> : ""}
+        </div>
+      );
+    }
```
Comment (stubailo) [inline examples/react-todos/client/components/accounts/UserSidebarSection.jsx]: Put else on the same line as closing bracket

File: examples/react-todos/client/components/ListShow.jsx [hunk 5]
```diff
@@ -126,81 +135,94 @@ ListShow = React.createClass({
...
-          return <TodoItem
-            key={ task._id }
-            task={ task }
-            beingEdited={ task._id === self.state.taskBeingEditedId }
-            onInitiateEdit={ self.setTaskBeingEdited.bind(self, task._id) }
-            onStopEdit={ self.setTaskBeingEdited.bind(self, null) }
-          />;
-        }) }
+    var showTodoItemHTML = self.data.tasks.map(function (task) {
+            return (
```
Comment (stubailo) [inline examples/react-todos/client/components/ListShow.jsx]: This is indented too far

File: examples/react-todos/client/components/AppBody.jsx [hunk 6]
```diff
@@ -19,24 +21,33 @@ setTimeout(function () {
   ShowConnectionIssues.set(true);
 }, CONNECTION_ISSUE_TIMEOUT);
 
+
+//
+// React classes.
```
Comment (stubailo) [inline examples/react-todos/client/components/AppBody.jsx]: Not sure this comment is necessary, I feel like `React.createClass` is pretty self-explanatory

File: examples/react-todos/client/components/AppBody.jsx [hunk 7]
```diff
@@ -5,6 +5,7 @@ var {
   RouteHandler
 } = ReactRouter;
 
+
```
Comment (stubailo) [inline examples/react-todos/client/components/AppBody.jsx]: I don't think two newlines here is necessary

**Comment by stubailo** [comment]
Looks like a good idea @giantelk, added some comments inline. Since this is mostly a code style improvement, my comments are a little nitpicky but I think it makes sense in this instance.

**Comment by giantelk** [comment]
I like white space, makes it easier to read IMHO. If you prefer without the white space, no sweat.

**Comment by stubailo** [comment]
@giantelk now that I think of it, and after googling around, maybe the preferred style is to add newlines: https://github.com/planningcenter/react-patterns#component-organization

Also, if you were to add doc comments for every method you would want newlines there. Let's go with your spacing, then.

File: examples/react-todos/client/components/accounts/UserSidebarSection.jsx (hunk 3 above)
Comment (stubailo) [inline examples/react-todos/client/components/accounts/UserSidebarSection.jsx]: Found one: https://github.com/planningcenter/react-patterns#component-organization

File: examples/react-todos/client/components/ListShow.jsx (hunk 5 above)
Comment (giantelk) [inline examples/react-todos/client/components/ListShow.jsx]: I fixed this with another Pull Request.

File: examples/react-todos/client/components/accounts/UserSidebarSection.jsx (hunk 4 above)
Comment (giantelk) [inline examples/react-todos/client/components/accounts/UserSidebarSection.jsx]: Fixed. btw, that's a error prone brace style as per Code Complete.

File: .gitignore [hunk 8]
```diff
@@ -1,2 +1,18 @@
 *~
 .DS_Store
+
+examples/react-todos/packages/npm-container/.npm/package/npm-shrinkwrap.json
```
Comment (stubailo) [inline .gitignore]: This should not be ignored, it's intentionally in the repository.

File: examples/react-todos/client/components/accounts/UserSidebarSection.jsx (hunk 3 above)
Comment (giantelk) [inline examples/react-todos/client/components/accounts/UserSidebarSection.jsx]: Read "Code Complete".

File: examples/react-todos/client/components/ListTodos.jsx (hunk 2 above)
Comment (giantelk) [inline examples/react-todos/client/components/ListTodos.jsx]: Fixed.

File: examples/react-todos/client/components/accounts/UserSidebarSection.jsx (hunk 3 above)
Comment (stubailo) [inline examples/react-todos/client/components/accounts/UserSidebarSection.jsx]: I have read a lot of parts of it, but AFAIK code complete doesn't have much to say about spacing inside React component definitions. Although I guess it just generalizes to "classes".

Great book, BTW

File: examples/react-todos/client/components/ListTodos.jsx (hunk 1 above)
Comment (giantelk) [inline examples/react-todos/client/components/ListTodos.jsx]: Fixed.

File: examples/react-todos/client/components/AppBody.jsx (hunk 6 above)
Comment (giantelk) [inline examples/react-todos/client/components/AppBody.jsx]: Your call, I like comments -)

File: .gitignore (hunk 8 above)
Comment (giantelk) [inline .gitignore]: I can't see how to undo this change in the GitHub for Mac client. It shows all my commit's except this. re: `.gitignore`

Comment (stubailo) [inline .gitignore]: OK, I can fix it manually before I merge.

**Comment by stubailo** [comment]
@giantelk one more thing I forgot, please sign here: https://contribute.meteor.com/

**Comment by giantelk** [comment]
Signed, Re: Contributing to Meteor.

**Comment by stubailo** [comment]
Merged as one commit: 4e494e1e3e388017c9b9a386d8d94b08d893d3a1

Thanks a lot!

**Pull request closed without merging**